### ISSUE TITLE
Finish refactoring the similarities task and add it to the pipeline

### ISSUE THREAD

**Issue by mclevey** [mclevey/podlm#28]
John - in progress already.

**Comment by mclevey** [comment]
@tcrick I know you've developed some really good vectorized `numpy` functions for doing cosine similarity comparisons. What I've developed is really performant, but we may still opt to use some of your functions, especially for some of the network-based stuff downstream. Do your functions already accept `graph-tool` inputs?

**Comment by tcrick** [comment]
Yep and we might want to GPU this process or use numba (or Polars).

Those cosine similarity functions predate our use of graph-tool but it shouldn't be too much of a problem to make the new versions work with properties from graph-tool objects.

I may have written some functions for this more recently, for diagnostics on the transformer embeddings in the CRS paper.
